### ISSUE TITLE
Allow setting "extends" via the Node.js API

### ISSUE THREAD

**Issue by feross** [eslint/eslint#5179]
The [`CLIEngine` constructor](http://eslint.org/docs/2.0.0/developer-guide/nodejs-api#cliengine) seems to be missing an `extends` option.

My workaround is to use `baseConfig`, like this:

```js
var CLIEngine = require("eslint").CLIEngine
var cli = new CLIEngine({
  baseConfig: {
    extends: [ 'standard' ]
  }
})
```

Is there a reason why this isn't exposed like `rules`, `globals`, `env`, etc.?

**Comment by eslintbot** [comment]
@feross Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by feross** [comment]
Also, `plugins` is missing too.

**Comment by nzakas** [comment]
The object passed to `CLIEngine` is not an ESLint config object, so it doesn't have the same fields.

* There's no such thing as `extends` on the CLI because there's no configuration to extend. You can use `configFile` to specify the same value as `-c` on the CLI or `baseConfig` to specify the base config to use, including `extends`.
* It looks like `plugins` is present. When you say "it's missing," what does that mean? Are you seeing an error? Is something not working as expected?

**Comment by feross** [comment]
> There's no such thing as `extends` on the CLI

Okay, fair enough. Though this means that it's not possible to invoke the CLI directly on a shared configuration without creating an extra file on the filesystem and referencing that with `-c`. I expected this to work:

```
npm install eslint eslint-config-standard
eslint --extends standard test.js
```

**Comment by feross** [comment]
`plugins` is missing from the [Node JS API documentation](http://eslint.org/docs/2.0.0/developer-guide/nodejs-api) for v2.0.0.

**Comment by ilyavolodin** [comment]
@feross I don't think `CLIEngine` supports plugins directly. You should use [addPlugin](http://eslint.org/docs/2.0.0/developer-guide/nodejs-api#addplugin) method instead.

**Comment by feross** [comment]
@ilyavolodin Ah, missed that somehow. Thanks.

**Comment by platinumazure** [comment]
@ilyavolodin Related, it seems pointless that `CLIEngine` excepts those plugins on its constructor but that it doesn't actually do anything unless you call `engine.addPlugin()`. Any reason why that option is even on the constructor in the first place?

**Comment by ilyavolodin** [comment]
I don't think it's in the constructor. Constructor takes baseConfig, which is config object that can accept `plugins` among other things (or at least under regular conditions it can). Problem is, since `baseConfig` isn't processed by config-file, plugins will not be loaded. But documentation doesn't specify that you can supply `plugins` as part of `baseConfig` and have them be loaded (however, it doesn't specify otherwise either, so that might need to be fixed).

**Comment by nzakas** [comment]
It is accepted as part of the constructor options. So you can do:

```js
var engine = new CLIEngine({ plugins: ["foo"] });
```

Those get passed through to [Config](https://github.com/eslint/eslint/blob/master/lib/cli-engine.js#L492), which uses it to [load plugins](https://github.com/eslint/eslint/blob/master/lib/config.js#L254).

So, looks like it's just missing from the docs but actually does work.

**Comment by platinumazure** [comment]
I'm not sure it does. I had to call addPlugin to make my plugin's
integration tests work. But this issue has gone off track, so I will open a
new issue if I can reproduce again.
